lib/utils/string_utils.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class StringUtils
+  class << self
+    def excerpt(text, kword, length)
+      txt = text.tr "\n", ''
+      idx = txt.downcase.index(/#{kword}/i)
+      pad = (length / 2) - 3
+      tmp = "...#{txt[[idx - pad, 0].max..idx + pad - 1]}..."
```

Comment:
This is another place where some searches throw an error, when `idx` is nil.

This whole method is also hard to understand on a quick read. What does `idx` mean here?

Reply:
I rewrote a bit StringUtils, added some checks + tests.